### ISSUE TITLE
Ionic 2 Android build fails when plugin is added to project

### ISSUE THREAD

**Issue by pdrosos** [mauron85/cordova-plugin-background-geolocation#195]
## Your Environment

- Plugin version: 2.2.4
- Platform: Android
- OS version: Linux Mint 18
- Device manufacturer and model: -
- Cordova version (`cordova -v`): 6.4.0
- Cordova platform version (`cordova platform ls`):
  Installed platforms:
  android 6.0.0
  Available platforms: 
  amazon-fireos ~3.6.3 (deprecated)
  blackberry10 ~3.8.0
  browser ~4.1.0
  firefoxos ~3.6.3
  ubuntu ~4.3.4
  webos ~3.7.0
- Plugin configuration options: -
- Link to your project: -

Issue details posted originally here: https://github.com/driftyco/ionic-native/issues/752 and Ionic team says it's this plugin's issue.

When I generate a new starter project:
`ionic start test-project --v2`
and add the [BackgroundGeolocation](http://ionicframework.com/docs/v2/native/backgroundgeolocation/) plugin from Ionic Native:
`ionic plugin add cordova-plugin-mauron85-background-geolocation`
the android build:
`ionic build android`
fails with this error:

```
:processDebugResources
/home/pavlina/Projects/ionic-2.rc-1/platforms/android/res/xml/authenticator.xml:5:19-33: AAPT: No resource found that matches the given name (at 'icon' with value '@drawable/icon').

/home/pavlina/Projects/ionic-2.rc-1/platforms/android/res/xml/authenticator.xml:6:24-38: AAPT: No resource found that matches the given name (at 'smallIcon' with value '@drawable/icon').

/home/pavlina/Projects/ionic-2.rc-1/platforms/android/build/intermediates/res/merged/debug/xml/authenticator.xml:2: error: Error: No resource found that matches the given name (at 'icon' with value '@drawable/icon').

/home/pavlina/Projects/ionic-2.rc-1/platforms/android/build/intermediates/res/merged/debug/xml/authenticator.xml:2: error: Error: No resource found that matches the given name (at 'smallIcon' with value '@drawable/icon').

FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processDebugResources'.
> com.android.ide.common.process.ProcessException: Failed to execute aapt

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 9.847 secs

Error: /home/pavlina/Projects/ionic-2.rc-1/platforms/android/gradlew: Command failed with exit code 1 Error output:
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
/home/pavlina/Projects/ionic-2.rc-1/platforms/android/res/xml/authenticator.xml:5:19-33: AAPT: No resource found that matches the given name (at 'icon' with value '@drawable/icon').

/home/pavlina/Projects/ionic-2.rc-1/platforms/android/res/xml/authenticator.xml:6:24-38: AAPT: No resource found that matches the given name (at 'smallIcon' with value '@drawable/icon').

/home/pavlina/Projects/ionic-2.rc-1/platforms/android/build/intermediates/res/merged/debug/xml/authenticator.xml:2: error: Error: No resource found that matches the given name (at 'icon' with value '@drawable/icon').

/home/pavlina/Projects/ionic-2.rc-1/platforms/android/build/intermediates/res/merged/debug/xml/authenticator.xml:2: error: Error: No resource found that matches the given name (at 'smallIcon' with value '@drawable/icon').

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processDebugResources'.
> com.android.ide.common.process.ProcessException: Failed to execute aapt

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
```

Please let me know how to fix this, as the plugin is not usable for me at the moment and I need it in my project. 

Thanks!

**Comment by mauron85** [comment]
Sorry I don't use ionic myself, so cannot be very helpful here. However I would say if you can add icon into some simlar path to `android/plugin/src/main/res/drawablei/icon.png`. It should work. Cordova is doing this as part of install process.

**Comment by R2D221** [comment]
This bug affects me too, and I'm using plain Cordova, version 6.4.0. Adding the image at /drawable/icon.png does solve it, but the plugin itself should do it.

**Comment by mauron85** [comment]
@R2D221 actually plugin should be doing it. cordova itself put icons into drawable-* for all screen sizes. So it is not actually in drawable but in drawable-hdpi, drawable-mdpi ... My advice was to put into drawable, because it's the easiest way how to validate issue. But normally you should have icons in drawable-* folders.

**Comment by R2D221** [comment]
Well, there's a bug somewhere, because my generated project doesn't have any drawable-hdpi etc. folders.

**Comment by pdrosos** [comment]
@mauron85 I have these drawable-* directories, but only putting icon in platforms/android/res/drawable/icon.png solves the issue. And strangest thing is I was able to build without issues only a couple of days ago...

**Comment by mauron85** [comment]
Just did test on cordova 6.4.0

`cordova create myApp org.apache.cordova.myApp myApp`
`cd myApp`
`cordova platform add android'

`find . -name drawable*`

```
./platforms/android/res/mipmap-xhdpi/icon.png
./platforms/android/res/mipmap-mdpi/icon.png
./platforms/android/res/mipmap-ldpi/icon.png
./platforms/android/res/mipmap-hdpi/icon.png
```

So it seems they changed from drawables to mipmap. Need to fix this.

**Comment by thanhlq** [comment]
Hello, so anyone has a workaround to bypass this issue? 

Thank you very much!

**Comment by thanhlq** [comment]
I confirm that, adding folder drawable/icon.png will solve problem :) : platforms/android/res/drawable/icon.png

**Comment by SPR-JoshLamkin** [comment]
Why is this Closed? You should be updating your authenticator.xml file to read the mipmap icons instead of the drawable. @mipmap/icon.png

**Comment by SPR-JoshLamkin** [comment]
I opened a Pull Request that fixes this issue.

**Comment by mauron85** [comment]
It is not closed. The ionic issue is closed, not this one.

**Comment by mauron85** [comment]
fixed in 2.2.5. thanks to @SPR-JoshLamkin
